tests/test_peak_properties.py

```diff
@@ -0,0 +1,98 @@
+import strax
+import numpy as np
+from hypothesis import given, strategies, example, settings
+import tempfile
+
+
+@settings(max_examples=100, deadline=None)
+@given(
+    # number of peaks
+    strategies.integers(min_value=0, max_value=20),
+    # length of the data field in the peaks
+    strategies.integers(min_value=2, max_value=20),
+)
+def test_index_of_fraction(peak_length, data_length):
+    """
+    Test strax.index_of_fraction
+    """
+    # Setup the peaks
+    dtype = [(('Start time since unix epoch [ns]', 'time'), np.int64),
+             (('Exclusive end time since unix epoch [ns]', 'endtime'), np.int64),
+             (('length of p', 'length'), np.int16),
+             (('area of p', 'area'), np.float64),
+             (('data of p', 'data'), (np.float64, data_length))]
+    peaks = np.zeros(peak_length, dtype=dtype)
+    peaks['time'] = np.arange(peak_length)
+    peaks['endtime'] = peaks['time'] + 1
+
+    # Fill the peaks with random length data
+    for p in peaks:
+        length = np.random.randint(0, data_length)
+        p['length'] = length
+        wf = np.random.random(size=data_length)
+        p['data'] = wf
+
+        for i in range(length, len(wf)):
+            # Set all the out of length data to zero
+            wf[i] = 0.

```

Comment:
nope, I also found that strange. Perhaps my syntax was off or its because it's unjitted but yeah

Reply:
Hmm okay this odd, because I was not sure and so I tried the following in a notebook: 

```
peaks = np.zeros(2, dtype=strax.peak_dtype())
for p in peaks:
    length = int(np.random.uniform(0, 20))
    data = np.random.uniform(0, 20, length)
    p['data'][:length] = data
```

Reply:
Ok, so the `Perhaps my syntax was off ` seems quite likely. Will change, thanks Daniel!